### PR TITLE
graphclust alignment support + RNAalifold multiformat

### PR DESCRIPTION
**Pull request by mmiladi**

### CODE REVIEW

**Comment by bgruening** [comment]
Is the python file missing?

**Comment by mmiladi** [comment]
Python3 grrrr

**Comment by mmiladi** [comment]
Next time I promise to separate PRs for different tools ;)

File: tools/GraphClust/Preprocessing/preprocessing.xml [hunk 1]
```diff
@@ -36,7 +46,8 @@
         <data name="data.names" format="txt" from_work_dir="FASTA/data.names" label="data.names"/>
         <data name="data.fasta.scan" format="fasta" from_work_dir="FASTA/data.fasta.scan" label="data.fasta.scan"/>
         <data name="FASTA" format="zip" from_work_dir="FASTA.zip" label="FASTA.ZIP"/>
-        <data name="shape_data_split" format="txt" from_work_dir="shape_data_split.react" label="SHAPE data splited"/>
+        <data name="shape_data_split" format="txt" from_work_dir="shape_data_split.react" label="SHAPE data split"/>
+        <data name="alignment_data_split" format="txt" from_work_dir="alignment_data_split.stk" label="Alignments data file split"/>
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/preprocessing.xml]: is the format here stockholm?

File: tools/GraphClust/Preprocessing/splitStockholm.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+seq_id = []
+seq_string = []
+orig_id = []
+name_file = "FASTA/data.names"
+array_all_chunks = []
+with open(name_file, 'r') as f:
+    content = f.read()
+    lines = content.split('\n')[:-1]
+    for line in lines:
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/splitStockholm.py]: `for line in f` should be more efficient ..

File: tools/GraphClust/Preprocessing/splitStockholm.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        seq_id.append(int(line.split()[0]))
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/splitStockholm.py]: You should do the `line.split()` only once and then access the data in the following lines.

File: tools/GraphClust/Preprocessing/splitStockholm.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        seq_string.append(line.split()[1])
+        orig_id_srt = line.split()[3]
+        orig_id_srt = orig_id_srt.rsplit('_',1)[0]
+        orig_id.append(orig_id_srt)
+
+
+
+with open(stk_file) as stk_in, open(target_f, 'w') as out_stk_handle:
+    alignments = AlignIO.parse(stk_in, "stockholm")#, alphabet=IUPAC.ambiguous_rna)
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/splitStockholm.py]: out_stk_handle is not used.

File: tools/rna_tools/vienna_rna/rnaalifold.xml [hunk 5]
```diff
@@ -92,7 +104,8 @@
 ]]>
     </command>
     <inputs>
-        <param format="txt" name="input" type="data" label="Clustal file"/>        
+        <param format="txt" name="input" type="data" label="Clustal/AlignmentFasta/Stockholm/MAF file"/>
```
Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: this format should include `maf,stockholm,fasta,clustalw` ... and so on.

File: tools/GraphClust/Preprocessing/preprocessing.xml (hunk 1 above)
Comment (mmiladi) [inline tools/GraphClust/Preprocessing/preprocessing.xml]: Does stockholm exist as a defined format?

Comment (bgruening) [inline tools/GraphClust/Preprocessing/preprocessing.xml]: https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample#L484

File: tools/GraphClust/Preprocessing/splitStockholm.py (hunk 3 above)
Comment (mmiladi) [inline tools/GraphClust/Preprocessing/splitStockholm.py]: Yup, sure

**Comment by mmiladi** [comment]
@bgruening thanks much! pls don't merge yet, few tweaks to alifold is on the way.

File: tools/rna_tools/vienna_rna/rnaalifold.xml [hunk 6]
```diff
@@ -112,7 +114,17 @@
                 <option value="0">0: Deactivates the calculation of the pair probabilities, saving about 50% in runtime</option>
                 <option value="1" >1: Calculate the partition function and base pairing probability matrix</option>				
             </param>
-            <param name="mea" type="float" value="1.0" label="Gamma Value" help="Calculate an MEA (maximum expected accuracy) structure. The parameter gamma tunes the importance of correctly predicted pairs versus unpaired bases. Thus, for small values of gamma the MEA structure will contain only pairs with very high probability. where the expected accuracy is computed from the pair probabilities: each base pair (i,j) gets a score 2*gamma*p_ij and the score of an unpaired base is given by the probability of not forming a pair." argument="--MEA"/>
+
+            <conditional name="measelect">
```
Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: should we add a test for this as well?

Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: mh, wait, we can simply make this float value optional, isn't it ... then we don't need the conditional ...

Comment (mmiladi) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: The argument parsing for alifold is confusing here. `--MEA` means mea is enabled with default gamma value 1.0. `--MEA=1.2` means mea is enabled and gamma is changed to 1.2.

Comment (mmiladi) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: Now the wrapper is consistent with RNAfold. I'll add a small test here

Comment (mmiladi) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: What is wrong with this test to turn MEA on?
```
<test>
            <param name="input" value="rnaalifold_input1.clustal"/>
            <param name="algorithm_options.measelect" value="true"/>
            <param name="algorithm_options.measelect.mea" value="yes"/>
            <param name="algorithm_options.measelect.meavalue" value="1.0"/>
            <output name="output" file="rnaalifold_result_MEA.txt"/>
        </test>
        
```

Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: try just `<param name="mea" value="yes"/>`

Comment (mmiladi) [inline tools/rna_tools/vienna_rna/rnaalifold.xml]: :+1:

**Comment by mmiladi** [comment]
is perl-threaded for bioconda discontinued ?

**Comment by bgruening** [comment]
it's just `perl` nowadays.

**Comment by mmiladi** [comment]
so `perl >=5.22.0` instead of `perl-threaded >=5.22.0` ?

**Comment by bgruening** [comment]
yes, the linter should tell you more details.

**Comment by mmiladi** [comment]
I only know of `conda build` :D

**Comment by mmiladi** [comment]
Looks good?

**Comment by bgruening** [comment]
@mmiladi please increase the tool version if you bump the requirements.

**Comment by mmiladi** [comment]
Thanks much Bjoern.

**Comment by bgruening** [comment]
Thanks to you. TS is already updated.

**Pull request merged**